pkg/api/api.go

```diff
@@ -68,6 +76,86 @@ func (a *ScratchDataAPIStruct) AuthGetDatabaseID(ctx context.Context) int64 {
 	return ctx.Value("databaseId").(int64)
 }
 
+func (a *ScratchDataAPIStruct) CreateQuery(w http.ResponseWriter, r *http.Request) {
+	apiKey := r.URL.Query().Get("api_key")
+	if apiKey != "local" { // Validate API key
+		w.WriteHeader(http.StatusUnauthorized)
+		w.Write([]byte("Unauthorized"))
+		return
+	}
+
+	var requestBody struct {
+		Query    string `json:"query"`
+		Duration int    `json:"duration"` // Duration in seconds
+	}
+
+	if err := json.NewDecoder(r.Body).Decode(&requestBody); err != nil {
+		w.WriteHeader(http.StatusBadRequest)
+		w.Write([]byte("Invalid request body"))
+		return
+	}
+
+	// Validate the query
+	if requestBody.Query == "" {
+		w.WriteHeader(http.StatusBadRequest)
+		w.Write([]byte("Query cannot be empty"))
+		return
+	}
+
+	// Generate a new UUID
+	queryUUID := uuid.New()
+
+	// Store the query and its expiration time
+	queryExpiration := time.Duration(requestBody.Duration)
+	a.cache.Set(queryUUID.String(), []byte(requestBody.Query), &queryExpiration)
+
+	// Return the UUID representing the query
+	w.Header().Set("Content-Type", "application/json")
+	response := struct {
+		QueryUUID string `json:"query_uuid"`
+	}{
+		QueryUUID: queryUUID.String(),
+	}
+	json.NewEncoder(w).Encode(response)
+}
+
+func (a *ScratchDataAPIStruct) executeQueryAndStreamData(w http.ResponseWriter, ctx context.Context, query string, databaseID int64, format string) error {
```

Comment:
Thank you for creating a new function. Can you update the `Select()` function to also use this? See #116 under "Endpoint to Query Data"

> You should refactor the existing [Select endpoint](https://github.com/scratchdata/scratchdata/blob/main/pkg/api/data.go#L14) in order to reuse code for streaming data to the client